controllers/openstackplaybookgenerator_controller.go

```diff
@@ -495,6 +508,29 @@ func (r *OpenStackPlaybookGeneratorReconciler) setCurrentState(instance *ospdire
 
 }
 
+func (r *OpenStackPlaybookGeneratorReconciler) syncConfigVersions(instance *ospdirectorv1beta1.OpenStackPlaybookGenerator, configVersions map[string]ospdirectorv1beta1.OpenStackConfigVersion) error {
+
+	for _, version := range configVersions {
+
+		// Check if this ConfigVersion already exists
+		foundVersion := &ospdirectorv1beta1.OpenStackConfigVersion{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: version.Name, Namespace: instance.Namespace}, foundVersion)
+		if err == nil {
+			//FIXME(dprince): update existing?
+		} else if err != nil && k8s_errors.IsNotFound(err) {
+			r.Log.Info("Creating a ConfigVersion", "ConfigVersion.Namespace", instance.Namespace, "ConfigVersion.Name", version.Name)
+			err = r.Client.Create(context.TODO(), &version)
+			if err != nil {
+				return err
+			}
+		} else if err != nil {
+			return err
+		}
+	}
+	//FIXME(dprince): remove deleted?
```

Comment:
Perhaps the same consideration here as with regards to the finalizer conversation above